discovery-infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -91,8 +99,40 @@ def format_disk(disk_path):
         # Fix for libvirt 6.0.0
         if image_size.isdigit():
             image_size += "G"
-        command = f'qemu-img create -f qcow2 {disk_path} {image_size}'
-        utils.run_command(command, shell=True)
+
+        cls.create_disk(disk_path, image_size)
+
+    def attach_disk(self, node, disk_size):
+        node = self.libvirt_connection.lookupByName(node)
+
+        # <target> dev attribute needs some unique device identifier for the disk (target_dev).
+        # This path will not be the same path used by the guest for this disk.
+        # See https://libvirt.org/formatdomain.html#elementsDisks:~:text=The%20target%20element%20controls%20the%20bus
+        # We generate a random one with a lot of digits so collision is very unlikely
+        # When detaching, we use the alias rather than target_dev,
+        # (unlike virsh attach-disk and detach-disk commands)
+        disk_id = random.randrange(int(1e12))
```

Comment:
Use `uuid.uuid4()`

Reply:
I need decimal digits without hyphens or hex... vda1892379817239 vda1237818723 etc

Reply:
```
In [11]: x                                                                      
Out[11]: UUID('ceb3ed14-bc4f-45a0-9294-b3cccc78f431')

In [12]: x.hex                                                                  
Out[12]: 'ceb3ed14bc4f45a09294b3cccc78f431'

In [13]: x.int                                                                  
Out[13]: 274755196843288735486359503772499637297
```

Reply:
While checking to see if such long numbers are supported (truncating UUID's is a bad idea) I stumbled upon a potential limitation/bug that makes this entire line irrelevant, so resolving for now.

/hold